src/Keboola/StorageApi/Metadata.php

```diff
@@ -112,6 +114,29 @@ public function postTableMetadata($tableId, $provider, $metadata)
         ));
     }
 
+    /**
+     * @param string $tableId
+     * @param string $provider
+     * @param array $metadata [['key' => 'some-key', 'value' => 'Some value'], ...]
+     * @param array $columnsMetadata ['columnName1' => [['key' => 'some-key', 'value' => 'Some value'], ...], ...]
+     * @return mixed|string
+     * @throws ClientException
+     */
+    public function postTableMetadataWithColumns($tableId, $provider, $metadata, $columnsMetadata)
+    {
+        if (!is_array($metadata) || count($metadata) === 0) {
+            throw new ClientException("Third argument must be a non-empty array of Metadata objects");
+        }
+        if (!is_array($columnsMetadata) || count($columnsMetadata) === 0) {
+            throw new ClientException("Fourth argument must be a non-empty array of Metadata objects with columns names as keys");
+        }
```

Comment:
úplně sa nestotožňuju stím, že když je to prázdné pole tak by měla failnůt validace na straně serveru.

Reply:
Tak je to stejny jako ten 3. argument. Taky kdyz uz ho tam das, tak by nemel byt prazdnej...